instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/JmxMetricInsight.java

```diff
@@ -33,7 +37,28 @@ private JmxMetricInsight(OpenTelemetry openTelemetry, long discoveryDelay) {
     this.discoveryDelay = discoveryDelay;
   }
 
-  public void start(MetricConfiguration conf) {
+  /**
+   * Starts metric registration for local JVM
+   *
+   * @param conf metric configuration
+   */
+  public void startLocal(MetricConfiguration conf) {
+    start(conf, () -> MBeanServerFactory.findMBeanServer(null));
+  }
+
+  /**
+   * Starts metric registration for a remote JVM connection
+   *
+   * @param conf metric configuration
+   * @param connections supplier for list of remote connections
+   */
+  public void startRemote(
```

Comment:
could you add a comment about why it takes `Supplier`? (e.g. is lazy initialization needed? or are different return values over time needed? or both?) thanks

Reply:
Good point, here it's needed for delayed initialization which is required as otherwise triggering JMX initialization early in `discoverBeans` breaks jboss/wildfly.

Reply:
Fixed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12178/commits/25cd7c0729e5a061aae8e17d21bf2db6cf43c2bc